packages/notebook/src/browser/view/notebook-cell-editor.tsx

```diff
@@ -0,0 +1,45 @@
+// *****************************************************************************
+// Copyright (C) 2023 TypeFox and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import * as React from '@theia/core/shared/react';
+import { NotebookModel } from '../view-model/notebook-model';
+import { NotebookCellModel } from '../view-model/notebook-cell-model';
+import { MonacoEditorProvider } from '@theia/monaco/lib/browser/monaco-editor-provider';
+
+interface EditorProps {
+    notebookModel: NotebookModel,
+    editorProvider: MonacoEditorProvider,
+    cell: NotebookCellModel
+}
+
+export function Editor({ notebookModel, editorProvider, cell }: EditorProps): JSX.Element {
+    const uri = cell.uri;
+    React.useEffect(() => {
+        (async () => {
+            const editorNode = document.getElementById(uri.toString())!;
+            const editor = await editorProvider.createInline(uri, editorNode, { minHeight: -1, maxHeight: -1, autoSizing: true }, true);
```

Comment:
Upon further inspection (I've installed the Pyright vscode extension) there are more issues with the editor:

1. There is no context menu (expected when using `createInline`, as it disables the context menu)
2. Opening inline widgets (such as the "find references" widget) inside the editor breaks the layout:
![grafik](https://user-images.githubusercontent.com/4377073/233088130-629dc700-c0b3-4891-a6fc-b6cdd3bc3bbf.png)

I guess we don't have time to fix these, so we will ignore the editor issues for know. These comments are for bookkeeping and documentation purposes.

Reply:
There are more issues with the Python/Pyright integration, but they're unrelated to our implementation. VSCode has the same issues when trying to run python notebooks with just Pyright.

Reply:
> Saving the notebook editor seems to only works in parts. The notebook is saved to disk, but without any formatting (it should use the normal JSON formatting) and the widget is still marked as being dirty.

the unformatted saving is probably because my extension is doing the bare minimum for serializing the notebook-data. This was mostly just for showing it works. the still dirty widget sounds like a bug